jax/experimental/jax2tf/tests/shape_poly_test.py

```diff
@@ -1189,209 +1374,189 @@ def f_jax(x):
+    check_shape_poly(self,
+                     jax.vmap(f_jax),
+                     arg_descriptors=[RandArg((3,), _f32)],
+                     input_signature=[tf.TensorSpec((None, 3), dtype=tf.float32)],
+                     polymorphic_shapes=["b, ..."],
+                     expected_output_signature=tf.TensorSpec((None, 3), dtype=tf.float32)
+                     )
 
-  `expect_error` is a pair of an Exception type and a regular expression to
-  match the expected exception string.
+  def test_reshape_compiled(self):
+    # We compile the result of conversion for two shapes, hence we need to
+    # involve the TF compiler twice, but we trace only once with shape polymorphism
+    traced = False
 
-  enable_and_disable_xla=True means that we generate two harnesses,
-  one with enable_xla=False and one with enable_xal=True. Otherwise we create
-  only one harness with enable_xla=True.
-  """
-  if enable_and_disable_xla:
-    return [
-        _make_harness(group_name, name + ("" if enable_xla else "_noxla"),  # type: ignore
-                      func, args, poly_axes=poly_axes,
-                      check_result=check_result, tol=tol, enable_xla=enable_xla,
-                      enable_and_disable_xla=False, skip_jax_run=skip_jax_run,
-                      expect_error=expect_error)
-        for enable_xla in [True, False]
-    ]
-  poly_axes_name = f"poly_axes={repr(poly_axes)}"
-  assert isinstance(poly_axes, Sequence)
-  # Make poly_axes: Sequence[Sequence[int]]
-  poly_axes = tuple(map(lambda pa: pa if isinstance(pa, Sequence) or pa is None else (pa,),
-                        poly_axes))
-  if name:
-    name = f"{name}_{poly_axes_name}"
-  else:
-    name = poly_axes_name
-  return Harness(group_name,
-                 name,
-                 func, args,
-                 dtype=np.float32,
-                 poly_axes=poly_axes, check_result=check_result,
-                 skip_jax_run=skip_jax_run, expect_error=expect_error,
-                 tol=tol,
-                 **params)
+    def f_jax(x):
+      nonlocal traced
+      traced = True
+      y = jnp.sin(x)
+      return y.reshape([x.shape[0], -1])
 
+    x = self.rng().rand(4, 2, 3)
+    res_jax = f_jax(x)
+
+    traced = False
+    # If we get_concrete_function we trace once
+    f_tf = tf.function(
+        jax2tf.convert(f_jax, polymorphic_shapes=[PS("b", ...)]),
+        autograph=False,
+        jit_compile=True).get_concrete_function(
+            tf.TensorSpec([None, 2, 3], x.dtype))
+    self.assertTrue(traced)
+    traced = False
+    self.assertAllClose(res_jax, f_tf(x))
+    self.assertFalse(traced)  # We are not tracing again
+
+    x = self.rng().rand(6, 2, 3)
+    res_jax = f_jax(x)
+    traced = False
+
+    self.assertAllClose(res_jax, f_tf(x))
+    self.assertFalse(traced)  # We are not tracing again
 
-_f32 = np.float32
 
 # List containing either harnesses, or lists of harnesses
 _POLY_SHAPE_TEST_HARNESSES = [
-    _make_harness("add", "",
```

Comment:
Most of the rest are just to replace _make_harness with PolyHarness and reformating.